Output from updated tests:

```
tests/integration/test_azure_integrator.py::test_lb[public] 
--------------------------------------------------------------------------------------------------------------------------------- live log call ----------------------------------------------------------------------------------------------------------------------------------
INFO     test_azure_integrator:test_azure_integrator.py:30 Creating public LB
INFO     test_azure_integrator:test_azure_integrator.py:35 Waiting for LB
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [idle] active: Ready
  lb-consumer/0 [executing] waiting: waiting on provider
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
  lb-consumer/0 [idle] waiting: waiting on provider response
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [idle] active: Ready
  lb-consumer/0 [idle] active: 23.96.56.71
INFO     test_azure_integrator:test_azure_integrator.py:39 Verifying LB components
INFO     test_azure_integrator:test_azure_integrator.py:51 Confirming external access to http://23.96.56.71/
INFO     test_azure_integrator:test_azure_integrator.py:56 Confirming access from azure-integrator/0 to http://23.96.56.71/
INFO     test_azure_integrator:test_azure_integrator.py:56 Confirming access from lb-consumer/0 to http://23.96.56.71/
INFO     test_azure_integrator:test_azure_integrator.py:65 Cleaning up LB
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [idle] active: Ready
  lb-consumer/0 [executing] active: 23.96.56.71
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [idle] active: Ready
PASSED
tests/integration/test_azure_integrator.py::test_lb[internal] 
--------------------------------------------------------------------------------------------------------------------------------- live log call ----------------------------------------------------------------------------------------------------------------------------------
INFO     test_azure_integrator:test_azure_integrator.py:30 Creating internal LB
INFO     test_azure_integrator:test_azure_integrator.py:35 Waiting for LB
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [idle] active: Ready
  lb-consumer/0 [executing] waiting: waiting on provider
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
  lb-consumer/0 [idle] waiting: waiting on provider response
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [idle] active: Ready
  lb-consumer/0 [idle] active: 192.168.0.6
INFO     test_azure_integrator:test_azure_integrator.py:39 Verifying LB components
INFO     test_azure_integrator:test_azure_integrator.py:56 Confirming access from azure-integrator/0 to http://192.168.0.6/
INFO     test_azure_integrator:test_azure_integrator.py:56 Confirming access from lb-consumer/0 to http://192.168.0.6/
ERROR    test_azure_integrator:test_azure_integrator.py:62 Failed: assert 'nginx' in '  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dlo...0      0 --:--:--  0:02:10 --:--:--     0\ncurl: (28) Failed to connect to 192.168.0.6 port 80: Connection timed out\n'
INFO     test_azure_integrator:test_azure_integrator.py:65 Cleaning up LB
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [idle] active: Ready
  lb-consumer/0 [idle] active: 192.168.0.6
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [executing] active: Ready
INFO     juju.model:model.py:2257 Waiting for model:
  azure-integrator/0 [idle] active: Ready
FAILED
------------------------------------------------------------------------------------------------------------------------------- live log teardown --------------------------------------------------------------------------------------------------------------------------------
INFO     pytest_operator.plugin:plugin.py:218 Model status:

Unit                 Machine  Status       Message
lb-consumer/0*       1        waiting      waiting on provider
azure-integrator/0*  0        active       Ready

Machine  Series      Status
0        focal       running
1        focal       running
INFO     pytest_operator.plugin:plugin.py:238 Juju error logs:

machine-1: 10:59:22 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
machine-0: 10:59:24 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
unit-azure-integrator-0: 11:21:20 ERROR juju.api.watcher error trying to stop watcher: relation scope id "406b58ac-7fc4-4509-8170-4d527fa9c188:r#3#lb-consumer": settings 406b58ac-7fc4-4509-8170-4d527fa9c188:r#3#lb-consumer not found (not found)
unit-lb-consumer-0: 11:21:20 ERROR juju.api.watcher error trying to stop watcher: relation scope id "406b58ac-7fc4-4509-8170-4d527fa9c188:r#3#azure-integrator": settings 406b58ac-7fc4-4509-8170-4d527fa9c188:r#3#azure-integrator not found (not found)
unit-lb-consumer-0: 11:21:20 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "406b58ac-7fc4-4509-8170-4d527fa9c188:r#3#azure-integrator": settings 406b58ac-7fc4-4509-8170-4d527fa9c188:r#3#azure-integrator not found (not found)
unit-azure-integrator-0: 11:21:20 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "406b58ac-7fc4-4509-8170-4d527fa9c188:r#3#lb-consumer": settings 406b58ac-7fc4-4509-8170-4d527fa9c188:r#3#lb-consumer not found (not found)
unit-lb-consumer-0: 12:02:01 ERROR juju.api.watcher error trying to stop watcher: relation scope id "406b58ac-7fc4-4509-8170-4d527fa9c188:r#7#azure-integrator": settings 406b58ac-7fc4-4509-8170-4d527fa9c188:r#7#azure-integrator not found (not found)
unit-lb-consumer-0: 12:02:01 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "406b58ac-7fc4-4509-8170-4d527fa9c188:r#7#azure-integrator": settings 406b58ac-7fc4-4509-8170-4d527fa9c188:r#7#azure-integrator not found (not found)
unit-azure-integrator-0: 12:02:01 ERROR juju.api.watcher error trying to stop watcher: relation scope id "406b58ac-7fc4-4509-8170-4d527fa9c188:r#7#lb-consumer": settings 406b58ac-7fc4-4509-8170-4d527fa9c188:r#7#lb-consumer not found (not found)
unit-azure-integrator-0: 12:02:01 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "406b58ac-7fc4-4509-8170-4d527fa9c188:r#7#lb-consumer": settings 406b58ac-7fc4-4509-8170-4d527fa9c188:r#7#lb-consumer not found (not found)



==================================================================================================================================== FAILURES ====================================================================================================================================
_______________________________________________________________________________________________________________________________ test_lb[internal] ________________________________________________________________________________________________________________________________
Traceback (most recent call last):
  File "/home/johnsca/juju/envs/dev/charms/azure-integrator/tests/integration/test_azure_integrator.py", line 59, in test_lb
    assert "nginx" in output
AssertionError: assert 'nginx' in '  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dlo...0      0 --:--:--  0:02:10 --:--:--     0\ncurl: (28) Failed to connect to 192.168.0.6 port 80: Connection timed out\n'
============================================================================================================================ short test summary info =============================================================================================================================
FAILED tests/integration/test_azure_integrator.py::test_lb[internal] - AssertionError: assert 'nginx' in '  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dlo...0      0 --:--:--  0:02:10 --:--:--     0\ncu...
```

--- a/tests/integration/test_azure_integrator.py
+++ b/tests/integration/test_azure_integrator.py
@@ -52,7 +52,13 @@ async def test_lb(ops_test, list_lbs, list_public_ips, list_nsg_rules, visibilit
             r = requests.get(lb_url)
             assert r.status_code == 200
             assert "nginx" in r.text.lower()
-        for unit in (az_unit, lb_unit):
+        units = [az_unit]
+        if visibility == "public":
+            # Backends can never reach their own internal LBs, so self-connectivity can
+            # only be validated for a public LB. See bullet #3 on:
+            # https://docs.microsoft.com/en-us/azure/load-balancer/components#limitations
+            units.append(lb_unit)
+        for unit in units:
             log.info(f"Confirming access from {unit.name} to {lb_url}")
             data = await unit.run(f"curl -i '{lb_url}'")
             output = data.results.get("Stdout", data.results.get("Stderr", ""))

--- a/tox.ini
+++ b/tox.ini
@@ -24,8 +24,6 @@ commands = pytest -svv --tb native {posargs} tests/unit
 setenv =
     PYTHONBREAKPOINT=ipdb.set_trace
 deps =
-    # Until 2.8.6 is released
-    https://github.com/juju/python-libjuju/archive/master.zip#egg=juju
     pytest
     pytest-operator
     loadbalancer-interface

